lib/Sema/DerivedConformanceCodable.cpp

```diff
@@ -173,8 +173,13 @@ static bool validateCodingKeysEnum(DerivedConformance &derived,
         break;
 
       case DoesNotConform:
+        // for accurate diagnostic, we show TypeRepr if present
```

Comment:
It might be a good idea to rephrase this to provide better context. It could be something along the lines of:

```suggestion
        // We use a TypeLoc here so diagnostics can show the type
        // as written by the user in source if possible. This is useful
        // when the user has written an IUO type for example, since
        // diagnostics would show the type as 'T?' instead of 'T!' if
        // we use a Type instead.
```

Reply:
Thanks @theblixguy, will update at both the places.